src/plugins/layout.ts

```diff
@@ -91,54 +89,6 @@ export default class Layout implements IPlugin {
     return shadowNode.computeInfo();
   }
 
-  private processMultipleNodeEvents<T extends ILayoutEventInfo>(eventInfos: T[]) {
-    let eventsData: IEventData[] = [];
-    for (let i = 0; i < eventInfos.length; i++) {
-      let eventState = this.createEventState(eventInfos[i]);
-      eventsData.push({
-        type: this.eventName,
-        state: eventState
-      });
-    }
-    addMultipleEvents(eventsData);
-  }
-
-  private createEventState<T extends ILayoutEventInfo>(eventInfo: T): ILayoutState {
-    let node = eventInfo.node;
-    let shadowNode = this.shadowDom.getShadowNode(eventInfo.index);
```

Comment:
>let shadowNode = this.shadowDom.getShadowNode(eventInfo.index); [](start = 4, length = 63)

Changes in this file are required because now that we don't preserve information about removed nodes in the shadowDom, this would cause an error when creating event state for a removed node. We need to pass detached shadowdom nodes in a summary and use them directly. Once that adjustment is made, the follow-on simplifications in this file occur naturally.